src/mocks/handlers/letter.ts

```diff
@@ -60,38 +60,77 @@ const letterHandler = [
   http.get(baseURL('/api/letter/reception/:letterId'), async (req) => {
     await delay(300);
 
-    const mswCase = new URLSearchParams(window.location.search).get(
-      QUERY_STRINGS.mswCase,
-    );
-
-    if (mswCase === '400') {
-      return new HttpResponse(ERROR_RESPONSES.accessDeniedLetter, {
-        status: 400,
-      });
-    }
-
     const result: Reception = {
-      createdAt: '2024-02-13T16:50:44',
+      createdAt: '2024-02-17T16:50:44',
       letterId: Number(req.params.letterId),
+      letterTag: {
+        ageStart: 20,
+        ageRangeEnd: 50,
+        equalGender: true,
+      },
       senderNickname: '낯선 고양이123',
       receiverNickname: '낯선 강아지456',
       content: `${req.params.letterId} 번째 편지 입니다. 여기 거 다 남겨두고서 혹시겨두고서 혹시나 기대도 포기하려 하오 그대 부디 잘 지내시오 기나긴 그대 침묵을 이별로 받아두겠소 행여 이 맘 다칠까 근심은 접어두오 오 사랑한 사람이여 더 이상 못보아도 사실 그대 있음으로 힘겨운 날들을 견뎌 왔음에 감사하오 좋은 사람 만나오 사는 동안 날 잊고 사시오 진정 행복하길 바라겠소 이 맘만 가져가오 기나긴 그대 침묵을 이별로 받아두겠소 행여 이 맘 다칠까 근심은 접어두오 오 사랑한 사람이여 더 이상 못보아도 사실 그대 있음으로 힘겨운 날들을 견뎌 왔음에 감사하오`,
       worryType: 'BREAK_LOVE',
+      // imagePath: '이미지가 존재하지 않습니다.',
+      imagePath:
+        'https://cdn.pixabay.com/photo/2014/11/30/14/11/cat-551554_1280.jpg',
     };
 
-    return HttpResponse.json(result);
+    const status: number = 200;
+
+    switch (status) {
+      case 200:
+        return HttpResponse.json(result);
+      case 400:
+        return new HttpResponse(ERROR_RESPONSES.accessDeniedLetter, {
+          status: 400,
+        });
+      default:
+        break;
+    }
   }),
 
   http.patch(baseURL('/api/letter/reception/reply/:letterId'), async () => {
     await delay(1000);
 
-    return HttpResponse.json();
+    const status: number = 200;
+
+    switch (status) {
+      case 200:
+        return HttpResponse.json();
+      case 400:
+        return new HttpResponse(ERROR_RESPONSES.accessDeniedLetter, {
+          status: 400,
+        });
+      case 401:
+        return new HttpResponse(ERROR_RESPONSES.alreadyReplyExist, {
+          status: 400,
+        });
+      default:
+        break;
+    }
   }),
 
-  http.patch(baseURL('/api/letter/reception/toss/:letterId'), async () => {
+  http.patch(baseURL('/api/letter/reception/pass/:letterId'), async () => {
     await delay(1000);
 
-    return HttpResponse.json();
+    const status: number = 200;
+
+    switch (status) {
+      case 200:
+        return HttpResponse.json();
+      case 400:
+        return new HttpResponse(ERROR_RESPONSES.accessDeniedLetter, {
+          status: 400,
+        });
+      case 401:
+        return new HttpResponse(ERROR_RESPONSES.repliedLetterPass, {
+          status: 400,
+        });
```

Comment:
p5;

아하.. case 401로 나뉘어진 이유가 궁금했는데, 같은 400번대 응답이지만 에러 메시지가 다른 경우여서 그랬군요!

그런데 화면 입장에서는 case 400과 case 401이 모두 **에러 메시지를 토스트 UI 등에 보여줘서 사용자에게 알려준다** 라는 동일한 동작을 해야하는 것으로 보여서, 모킹에서 나누지 않아도 괜찮을 것 같다는 생각이 들어요. (400 케이스 하나만 놓기)

(달라야 한다면, 가령 401에는 사용자가 입력한 필드 타입 중 어떤 부분이 유효성 검사에 실패했는지 구체적인 체크가 들어가고, 400에는 단순 에러 메시지만 보여주는 등.. 화면에서의 동작이 완전 달라지는 경우일 것 같네요)

혹은 제가 놓친 케이스가 있었을지? 궁금해요

```suggestion
      case 400:
        return new HttpResponse(ERROR_RESPONSES.accessDeniedLetter, {
          status: 400,
        });
```